### ISSUE TITLE
[2.x] Prefetch doesn't load deferred props

### ISSUE THREAD

**Issue by javieralarcn10** [inertiajs/inertia#2108]
### Version:

- `@inertiajs/react` version: 2.0.0-beta.2

### Describe the problem:

When I prefetch a <Link prefetch/> it only loads props that do not have the Inertia::defer method. 

### Steps to reproduce:

Controller: 
`return inertia('Dashboard/Products/Inventory', [
            'shop' => $shop,
            'inventory' => Inertia::defer(fn() => $shop->inventory),
]);`
        
 Frontened:
 `<Link prefetch href={route("inventory")} className="flex gap-2 items-center">
   <span className="font-medium text-white">Inventory</span>
   </Link>`

**Comment by mreduar** [comment]
That is not the objective of defer? That they only load when you enter the page?

**Comment by RobertBoes** [comment]
I agree, it shouldn't prefetch deferred props, as that may take a long time while in the meantime the user could navigate, which would then cause the page to load very slowly

**Comment by mathewparet** [comment]
It should load deferred props too, but as a second request (like how prefetch works normally).

For example, I assume that if a user is hovering over Paginator links chances are they want to click the next page or the page number over which they are hovering.

This works fine except if the page has deferred components, it looks like prefetch does prefetch the initial data, but doesn't seem to go back for the deferred data. The problem with this is now that if a user clicks on a page then nothing shows at all.

Oh! BTW, this issue exists in Vue also.

**Comment by EmilMoe** [comment]
I agree with @mathewparet. At least there should be some kind of option to prefetch deferred 

```
prefetch="hover" :prefetch-deferred="['prop']"
```

Or something more suitable.

**Comment by darkamenosa** [comment]
One more note on this, cache-for also should work similar for non deferred props. So when the user click back, they see rendered data.

**Comment by pascalbaljet** [comment]
I'm closing this issue as deferred props are designed to be loaded after the subsequent page is loaded, so strictly speaking this is not a bug. I'm also not sure about the `prefetch-deferred` Link prop idea with the second request. Prefetch responses are usually quite fast, and you don't want network requests stacking up for all deferred prop groups for all prefetched pages.
